kernel/src/common/math.rs

```diff
@@ -49,6 +50,27 @@ pub fn closest_power_of_two(mut num: u32) -> u32 {
     num
 }
 
+#[derive(Copy,Clone,PartialEq,PartialOrd,Eq,Ord)]
+pub struct PowerOfTwo(u32);
+
+impl PowerOfTwo {
+    pub fn bit<R>(self) -> R
+        where R: From<u32>
+    {
+        From::from(self.0)
+    }
+
+    pub fn zero() -> PowerOfTwo {
+        PowerOfTwo(0)
+    }
+}
+
+impl<F: Into<u32>> From<F> for PowerOfTwo {
+    fn from(f: F) -> PowerOfTwo {
+        PowerOfTwo(log_base_two(f.into()))
```

Comment:
Right, I think being explicit is the right choice here. In particular, the grant region comes to mind, where I think ceiling will be more appropriate (following the same argument, where it's now _ceiling_ that will underexpose memory)